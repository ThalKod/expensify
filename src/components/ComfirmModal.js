import React from "react";
import Modal from "react-modal";

export default (props)=>{
    return (
        <Modal
        isOpen={props.showModal}
        contentLabel = "Comfirm Deletion"
        closeTimeoutMS={200}
        className="modal"
        >
            <h3>Are You sure you want to delete it ?</h3>
            <button onClick={props.removeExpense}>YES</button>
            <button onClick={props.handleCloseModal}>NO</button>
        </Modal>
    );
};