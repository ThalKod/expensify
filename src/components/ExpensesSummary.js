import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import selectExpenses from "../selectors/expenses";
import selectExpenseTotal from "../selectors/expenses-total";
import numeral from "numeral";

export const ExpensesSummary = (props)=>{
    const expenseWord = props.expenseCount > 1 ? "expenses" : "expense";
    const hiddenExpensesCount = props.totalNumberOfExpenses - props.expenseCount;
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{props.expenseCount}</span> {expenseWord} totalling <span>{numeral(props.expenseTotal /100).format("$0,0.00")}</span></h1>
                {
                    hiddenExpensesCount > 0 ? <p>Not showing {hiddenExpensesCount} expense(s) because of Filters</p> : <p>No filters in place</p>
                }
                <div className="page-header__action">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    )
};

const mapStatetoprops = (state)=>{
    const visibleExpenses = selectExpenses(state.expenses, state.filter);
    return {
        totalNumberOfExpenses: state.expenses.length,
        expenseCount: visibleExpenses.length,
        expenseTotal: selectExpenseTotal(visibleExpenses)
    };
};

export default connect(mapStatetoprops)(ExpensesSummary);