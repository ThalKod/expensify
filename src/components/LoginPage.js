import React from "react";
import { connect } from "react-redux";
import { startLoginGoogle, startLoginGithub } from "../actions/auth";

export class LoginPage extends React.Component {

    render(){
        const { startLoginGoogle, startLoginGithub } = this.props;

        return (
            <div className="box-layout">
                <div className="box-layout__box">
                    <h1 className="box-layout__title">Expensify</h1>
                    <p>Manage your expenses...</p>
                    <button className="button button--login" onClick={startLoginGoogle}>Login with Google</button>
                    <button className="button button--login" onClick={startLoginGithub}>Login with Github</button>
                </div>
            </div>
        );
    };
}

const mapDispatchToProps = (dispatch)=>({
    startLoginGoogle: () => dispatch(startLoginGoogle()),
    startLoginGithub: () => dispatch(startLoginGithub()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);