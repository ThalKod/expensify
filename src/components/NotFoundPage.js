import React from "react";
import { Link } from "react-router-dom";


const NotFoundPAge = () =>(
    <div>
        404! <Link to="/">Go Home</Link>
    </div>
);

export default NotFoundPAge;