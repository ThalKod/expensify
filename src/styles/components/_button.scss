
.login-button{
    color: white;
    background: blueviolet;
    border: none;
    padding: $s-size;
    font-weight: 300;
    font-size: $font-size-large;
}