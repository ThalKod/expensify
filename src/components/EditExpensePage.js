import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { editExpense, startRemoveExpense, startEditExpense } from "../actions/expenses";
import ComfirmModal from "./ComfirmModal";


class EditExpensePage extends React.Component {

    state = {
        showModal: false,
    };
    
    removeExpense = ()=>{
        this.props.dispatch(startRemoveExpense({ id: this.props.expense.id }))
        this.props.history.push("/");
    }

    handleCloseModal = ()=>{
        this.setState({ showModal: false });
    }

    render(){
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={(expense)=>{
                        this.props.dispatch(startEditExpense(this.props.expense.id,expense));
                        this.props.history.push("/");
                    }}
                    />
                    <button className="button button--secondary" onClick={()=>{
                        this.setState({ showModal: true });
                    }}>remove</button>
                </div>
                <ComfirmModal showModal={this.state.showModal} handleCloseModal={this.handleCloseModal} removeExpense={this.removeExpense} />
            </div>
        );
    }
}

const mapStateToProps = (state, props)=>{
    return {
        expense: state.expenses.find((expense)=> expense.id === props.match.params.id)
    }
};

export default connect(mapStateToProps)(EditExpensePage);