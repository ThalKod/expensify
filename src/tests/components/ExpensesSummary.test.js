import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";


test("should render ExpenseSummary with 1 count", ()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={195}  />);
    expect(wrapper).toMatchSnapshot();
});

test("should render ExpenseSummary with 2 count", ()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expenseTotal={295}  />);
    expect(wrapper).toMatchSnapshot();
});