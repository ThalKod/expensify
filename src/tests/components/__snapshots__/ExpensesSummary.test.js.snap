// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with 1 count 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $1.95
      </span>
    </h1>
    <div
      className="page-header__action"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpenseSummary with 2 count 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        2
      </span>
       
      expenses
       totalling 
      <span>
        $2.95
      </span>
    </h1>
    <div
      className="page-header__action"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
